### PR TITLE
[CONSVC-1855] test: migrate existing contract tests and integrate into CI

### PR DESCRIPTION
**Pull request by Trinaa**
## Description
* migrate framework code from [merino](https://github.com/mozilla-services/merino/tree/main/test-engineering/contract-tests) project 
* create `make contract-test` and `make contract-test-clean` shortcuts to execute and cleanup contract-tests
* add contract tests to CI

**NOTE**
* Contract tests are currently failing. client_variants and WikiFruit provider are currently unsupported.
* Next step: [CONSVC-1947](https://mozilla-hub.atlassian.net/browse/CONSVC-1947) - Align contract test tech stack with application

## Issue(s)
[CONSVC-1855](https://mozilla-hub.atlassian.net/browse/CONSVC-1855)

### CODE REVIEW

**Review by hackebrot — CHANGES_REQUESTED** [review]
Hi @Trinaa! 👋🏻 

I've added comments inline.

We can discuss options for resolving the failures you've mentioned on Slack.

File: Makefile [hunk 1]
```diff
@@ -30,3 +30,9 @@ dev: $(INSTALL_STAMP)
 
 run: $(INSTALL_STAMP)
 	$(POETRY) run uvicorn $(APP_DIR).main:app
+
+contract-test:
+	docker-compose -f test-engineering/contract-tests/docker-compose.yml -p merino-py-contract-tests up --abort-on-container-exit --build
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply Makefile]: change applied: [3bd00e0](https://github.com/mozilla-services/merino-py/pull/27/commits/3bd00e0f474dbf7f7590af14c6c32c6e78603a68)

File: README.md [hunk 2]
```diff
@@ -50,6 +50,13 @@ $ make dev
 
 # Run merino-py without the auto code reloading
 $ make run
+
+# Run contract tests
+$ make contract-test
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply README.md:55]: change applied: [3bd00e0](https://github.com/mozilla-services/merino-py/pull/27/commits/3bd00e0f474dbf7f7590af14c6c32c6e78603a68)

File: README.md [hunk 3]
```diff
@@ -50,6 +50,13 @@ $ make dev
...
+
+# Run contract tests cleanup
+$ make contract-test-clean
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply README.md:58]: change applied: [3bd00e0](https://github.com/mozilla-services/merino-py/pull/27/commits/3bd00e0f474dbf7f7590af14c6c32c6e78603a68)

File: test-engineering/contract-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+This directory contains source code for automated contract tests for Merino.
+
+**Please note that this is work in progress.** 🚧
+
+## Run tests locally
+
+You can run the contract tests locally from the repository root directory using:
+
+```text
+docker-compose -f test-engineering/contract-tests/docker-compose.yml up --abort-on-container-exit --build
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply test-engineering/contract-tests/README.md]: change applied: [3bd00e0](https://github.com/mozilla-services/merino-py/pull/27/commits/3bd00e0f474dbf7f7590af14c6c32c6e78603a68)

File: test-engineering/contract-tests/client/README.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+### Steps
+
+#### Kinto Service
+
+* The Kinto service scenario step populates the records used by the Merino service
+* Kinto `request` fields:
+  * `service` - Set the value to `kinto`, to direct requests to the Kinto service.
+  * `delay` - (optional) Set seconds to pause before execution of request.
+  * `filename` - Set the file with records to upload. The files are located in
+                 `..\contract-tests\volumes\kinto`.
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply test-engineering/contract-tests/client/README.md]: change applied: [3bd00e0](https://github.com/mozilla-services/merino-py/pull/27/commits/3bd00e0f474dbf7f7590af14c6c32c6e78603a68)

File: test-engineering/contract-tests/client/tests/exceptions.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+class KintoError(Exception):
+    """Error specific to Kinto service interactions."""
+
+    pass
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply test-engineering/contract-tests/client/tests/exceptions.py]: change applied: [3bd00e0](https://github.com/mozilla-services/merino-py/pull/27/commits/3bd00e0f474dbf7f7590af14c6c32c6e78603a68)

File: test-engineering/contract-tests/kinto-setup/README.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+* `KINTO_COLLECTION`: The ID of the Kinto collection to create
+
+## Contributing
+
+**mypy** is used for type checking. Execute from the root using command:
+
+```text
+mypy \
+test-engineering/contract-tests/kinto-setup/ \
+--config-file=test-engineering/contract-tests/kinto-setup/mypy.ini
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply test-engineering/contract-tests/kinto-setup/README.md]: change applied: [3bd00e0](https://github.com/mozilla-services/merino-py/pull/27/commits/3bd00e0f474dbf7f7590af14c6c32c6e78603a68)

**Review by ncloudioj — APPROVED** [review]
LGTM, thanks for setting up CT and migrating them over to here!

The broken tests should be fixed by #28. We can merge that onto this branch so CI will remain green.

**Review by hackebrot — APPROVED** [review]
Great work, @Trinaa! 🚀

**Pull request merged**